File: Task_3/task_3.kt
fun isNumberIncreasing(number: Int): Boolean {
    if (number / 10 < 1) return false
    val digits = mutableListOf<Int>()

    var n = number
    while (n > 0) {
        digits.add(n % 10)
        n /= 10
    }

    digits.reverse()
    for (i in 1 until digits.size) {
        if (digits[i] <= digits[i - 1]) return false
    }

    return true
}

fun reverseNumber(number: Int): Int {
    val digits = mutableListOf<Int>()
    var n = number
    while (n > 0) {
        digits.add(n % 10)
        n /= 10
    }

    var reversedNumber = 0
    for (digit in digits) {
        reversedNumber = reversedNumber * 10 + digit
    }
    return reversedNumber
}

fun main() {
    println("¬ведите числа, которые хотите проверить через пробел (Ctrl+D дл€ окончани€):")
    val numbers = mutableListOf<Int>()
    while (true) {
        val input = readLine() ?: break
        numbers.addAll(input.split(" ").mapNotNull { it.toIntOrNull() })
    }

    println()
    var countSuccess = 0
    for (n in numbers) {
        if (isNumberIncreasing(n)) {
            print("${reverseNumber(n)} ")
            countSuccess++
        }
    }

    if (countSuccess == 0) {
        println("»скомых чисел не найдено!")
    }
}

File: Task_3/task_3.py
def is_number_increasing(number):
    if number // 10 < 1:
        return False
    digits = []

    while number > 0:
        digits.append(number % 10)
        number //= 10

    digits.reverse()
    for i in range(1, len(digits)):
        if digits[i] <= digits[i - 1]:
            return False

    return True


def reverse_number(number):
    digits = []
    while number > 0:
        digits.append(number % 10)
        number //= 10

    reversed_number = 0
    for digit in digits:
        reversed_number = reversed_number * 10 + digit
    return reversed_number


def main():
    print("¬ведите числа, которые хотите проверить через пробел (Ctrl+D дл€ окончани€):")
    try:
        # —читываем все строки ввода до EOF (Ctrl+D)
        numbers = list(map(int, input().split()))
    except EOFError:
        numbers = []

    print()
    count_success = 0
    for n in numbers:
        if is_number_increasing(n):
            print(reverse_number(n), end=" ")
            count_success += 1

    if count_success == 0:
        print("»скомых чисел не найдено!")


if __name__ == "__main__":
    main()